org.eclipse.m2e.pde.ui/src/org/eclipse/m2e/pde/ui/target/editor/internal/part/DependencyEditorViewerPart.java

```diff
@@ -0,0 +1,214 @@
+		tableColumn.setEditingSupport(new MavenTextEditingSupport(viewer, model, getter, setter));
+
+		getter = MavenTargetDependency::getVersion;
+		setter = MavenTargetDependency::setVersion;
+		tableColumn = newGenericColumn(250, getter);
+		tableColumn.getColumn().setText(Messages.MavenTargetDependencyEditor_3);
+		tableColumn.setEditingSupport(new MavenTextEditingSupport(viewer, model, getter, setter));
+
+		getter = MavenTargetDependency::getClassifier;
+		setter = MavenTargetDependency::setClassifier;
+		tableColumn = newGenericColumn(125, getter);
+		tableColumn.getColumn().setText(Messages.MavenTargetDependencyEditor_4);
+		tableColumn.setEditingSupport(new MavenTextEditingSupport(viewer, model, getter, setter));
+
+		getter = MavenTargetDependency::getType;
+		tableColumn = newGenericColumn(125, getter);
+		tableColumn.getColumn().setText(Messages.MavenTargetDependencyEditor_5);
+		tableColumn.setEditingSupport(new MavenTypeEditingSupport(viewer, model));
+	}
+
+	private TableViewerColumn newGenericColumn(int weight, Function<MavenTargetDependency, String> getter) {
+		TableViewerColumn tableColumn = new TableViewerColumn(viewer, SWT.NONE);
+		tableColumn.getColumn().addSelectionListener(new TableColumnSortListener(viewer, getter));
+		tableColumn.setLabelProvider(new MavenTargetColumnLabelProvider<>(getter));
+		getLayout().setColumnData(tableColumn.getColumn(), new ColumnWeightData(weight, false));
+		return tableColumn;
+	}
+
+	/**
+	 * Registers a selection-change listener for the viewer. Any change to the
+	 * selection has to be propagated to the data model in order to properly handle
+	 * e.g. the {@code enabled} state of the {@code Remove} button.
+	 */
+	@SuppressWarnings("unchecked") // The TableViewer is using raw types...
+	private void newSelectionChangeListener() {
+		viewer.addSelectionChangedListener(event -> {
+			if (ignoreSelectionChanges) {
+				return;
+			}
+			model.setCurrentSelection(event.getStructuredSelection().toList());
+		});
+	}
+
+	/**
+	 * Registers a new model-change listener. Whenever the model has changed (e.g.
+	 * new values have been added), the viewer has to be updated in order to show
+	 * those changes.
+	 */
+	private void newModelChangeListener() {
+		ISideEffectFactory factory = WidgetSideEffects.createFactory(composite);
+
+		factory.create(this::onInputChange);
+		factory.create(this::onSelectionChange);
+	}
+
+	/**
+	 * Registers a new key listener. Whenever the {@link SWT#DEL} key is pressed,
+	 * the currently selected elements should be removed from the model.
+	 */
+	private void newKeyListener() {
+		viewer.getTable().addKeyListener(KeyListener.keyPressedAdapter(event -> {
+			if (event.keyCode == SWT.DEL && !viewer.getSelection().isEmpty()) {
+				model.remove();
+			}
+		}));
+	}
+
+	/**
+	 * Updates the input and selection of this viewer with the entries from the
+	 * model. Does nothing if the entries are already up-to-date.
+	 */
+	private void onInputChange() {
```

Comment:
This method and the following `onSelectionChange()` seem to be simple enough to be inlined, which would replace the method-reference by a lambda-block.